src/services/db/GitFileSystemService.ts

```diff
@@ -416,13 +416,20 @@ export default class GitFileSystemService {
   // Get the Git log of a particular branch
   getGitLog(
     repoName: string,
-    branchName: string
+    branchName: string,
+    maxCount?: number
```

Comment:
not a fan of exposing this flexibility when we no function needs this

Reply:
Aye, I hesitated on it 🤔 

The method is called `getGitLog()`, so I didn't want to hardcode the `1` in the command since getGitLog could legitimately be used to retrieve more commits if we wanted to (even if it's not the case now).

If I hardcoded the `max-count:1`, the method should be renamed to `getLatestCommit` (or similar), but we already have a method named `getLatestCommitOfBranch()`. I suppose alternative is to remove the `getGitLog()` entirely and put the git command into `getLatestCommitOfBranch()` directly as an internal step 🤔 

Any preference/suggestion?

Reply:
can we just give a default argument of `1`? i think we should have a sane default for this and only those who need extra power call it explicitly

Reply:
> can we just give a default argument of 1? i think we should have a sane default for this and only those who need extra power call it explicitly

Done: made `maxCount` mandatory with default value of 1. Having a guaranteed value simplifies the cli args composition too: https://github.com/isomerpages/isomercms-backend/pull/1243/commits/7bae2e7033194ae0f354220b228e3b352c7fd2c6

Also added some sanity validation and tests as well: https://github.com/isomerpages/isomercms-backend/pull/1243/commits/16dc076ffd935afbbdb4a6b82765c87322c1a5b

In term of exposing functionality, I think that concern can be dismissed for now: there are no other callers to `getGitLog()` and the worst that can happen is the behaviour that was there before (returning the entire commit list). I'm keen in pushing the perf fix asap, and further refactors can be done later if you want to remove a method that you judge to be unnecessary.